SDL2 testSavePNG32 fails

```
======================================================================
FAIL: testSavePNG32 (test.image_test.ImageModuleTest)
see if we can save a png with color values in the proper channels.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rene/dev/pygame/pygame/test/image_test.py", line 180, in testSavePNG32
    self.assertEquals(pixels_as_tuples[3], greyish_pixel)
AssertionError: Tuples differ: (65, 71, 76, 145) != (115, 125, 135, 145)

First differing element 0:
65
115

- (65, 71, 76, 145)
+ (115, 125, 135, 145)
```

--- a/buildconfig/Setup.SDL2.in
+++ b/buildconfig/Setup.SDL2.in
@@ -23,7 +23,7 @@ DEBUG =
 #everything you can, but you can ignore ones you don't have
 #dependencies for, just comment them out
 
-imageext src_c/imageext.c $(SDL) $(IMAGE) $(DEBUG)
+imageext src_c/imageext.c $(SDL) $(IMAGE) $(PNG) $(JPEG) $(DEBUG)
 font src_c/font.c $(SDL) $(FONT) $(DEBUG)
 mixer src_c/mixer.c $(SDL) $(MIXER) $(DEBUG)
 mixer_music src_c/music.c $(SDL) $(MIXER) $(DEBUG)

--- a/buildconfig/config_win.py
+++ b/buildconfig/config_win.py
@@ -419,8 +419,10 @@ def setup(sdl2):
         DEPS.add_dummy('PORTTIME')
         DEPS.add('MIXER', 'SDL2_mixer', ['SDL2_mixer-[1-9].*'], r'(lib){0,1}SDL2_mixer\.dll$',
                  ['SDL', 'vorbisfile'])
-        DEPS.add('PNG', 'png', ['libpng-[1-9].*'], r'(png|libpng)[-0-9]*\.dll$', ['z'])
-        DEPS.add('JPEG', 'jpeg', ['jpeg-[6-9]*'], r'(lib){0,1}jpeg[-0-9]*\.dll$')
+        DEPS.add('PNG', 'png', ['SDL2_image-[2-9].*', 'libpng-[1-9].*'], r'(png|libpng)[-0-9]*\.dll$', ['z'],
+                 find_header='png\.h', find_lib='(lib)?png1[-0-9]*\.lib')
+        DEPS.add('JPEG', 'jpeg', ['SDL2_image-[2-9].*', 'jpeg-9*'], r'(lib){0,1}jpeg-9\.dll$',
+                 find_header='jpeglib\.h', find_lib='(lib)?jpeg-9\.lib')
         DEPS.add('IMAGE', 'SDL2_image', ['SDL2_image-[1-9].*'], r'(lib){0,1}SDL2_image\.dll$',
                  ['SDL', 'jpeg', 'png', 'tiff'], 0)
         DEPS.add('FONT', 'SDL2_ttf', ['SDL2_ttf-[2-9].*'], r'(lib){0,1}SDL2_ttf\.dll$', ['SDL', 'z', 'freetype'])
@@ -471,8 +473,8 @@ def setup_prebuilt_sdl2(prebuilt_dir):
     png.path = imageDep.path
     png.inc_dir = [os.path.join(prebuilt_dir, 'include').replace('\\', '/')]
     png.found = True
-    jpeg = DEPS.add('JPEG', 'jpeg', ['SDL2_image-[2-9].*', 'jpeg-[6-9]*'], r'(lib){0,1}jpeg[-0-9]*\.dll$',
-                   find_header='jpeglib\.h', find_lib='(lib)?jpeg[-0-9]*\.lib')
+    jpeg = DEPS.add('JPEG', 'jpeg', ['SDL2_image-[2-9].*', 'jpeg-9*'], r'(lib){0,1}jpeg-9\.dll$',
+                   find_header='jpeglib\.h', find_lib='(lib)?jpeg-9\.lib')
     jpeg.path = imageDep.path
     jpeg.inc_dir = [os.path.join(prebuilt_dir, 'include').replace('\\', '/')]
     jpeg.found = True

--- a/buildconfig/config_win.py
+++ b/buildconfig/config_win.py
@@ -378,8 +378,6 @@ def _add_sdl2_dll_deps(DEPS):
     DEPS.add_dll(r'(lib)?opus[-0-9]*\.dll$', 'opus', ['*opus-[0-9]*'])
     DEPS.add_dll(r'(lib)?opusfile[-0-9]*\.dll$', 'opusfile', ['*opusfile-[0-9]*'])
     # IMAGE
-    DEPS.add_dll(r'(png|libpng.*)\.dll$', 'png', ['libpng-[1-9].*'], ['z'])
-    DEPS.add_dll(r'(lib){0,1}jpeg[-0-9]*\.dll$', 'jpeg', ['jpeg-[6-9]*'])
     DEPS.add_dll(r'(lib){0,1}tiff[-0-9]*\.dll$', 'tiff', ['tiff-[0-9]*'], ['jpeg', 'z'])
     DEPS.add_dll(r'(z|zlib1)\.dll$', 'z', ['zlib-[1-9].*'])
     DEPS.add_dll(r'(lib)?webp[-0-9]*\.dll$', 'webp', ['*webp-[0-9]*'])
@@ -421,6 +419,8 @@ def setup(sdl2):
         DEPS.add_dummy('PORTTIME')
         DEPS.add('MIXER', 'SDL2_mixer', ['SDL2_mixer-[1-9].*'], r'(lib){0,1}SDL2_mixer\.dll$',
                  ['SDL', 'vorbisfile'])
+        DEPS.add('PNG', 'png', ['libpng-[1-9].*'], r'(png|libpng)[-0-9]*\.dll$', ['z'])
+        DEPS.add('JPEG', 'jpeg', ['jpeg-[6-9]*'], r'(lib){0,1}jpeg[-0-9]*\.dll$')
         DEPS.add('IMAGE', 'SDL2_image', ['SDL2_image-[1-9].*'], r'(lib){0,1}SDL2_image\.dll$',
                  ['SDL', 'jpeg', 'png', 'tiff'], 0)
         DEPS.add('FONT', 'SDL2_ttf', ['SDL2_ttf-[2-9].*'], r'(lib){0,1}SDL2_ttf\.dll$', ['SDL', 'z', 'freetype'])
@@ -465,6 +465,17 @@ def setup_prebuilt_sdl2(prebuilt_dir):
     ]
     ftDep.inc_dir.append('%s/freetype2' % ftDep.inc_dir[0])
     ftDep.found = True
+
+    png = DEPS.add('PNG', 'png', ['SDL2_image-[2-9].*', 'libpng-[1-9].*'], r'(png|libpng)[-0-9]*\.dll$', ['z'],
+                   find_header='png\.h', find_lib='(lib)?png1[-0-9]*\.lib')
+    png.path = imageDep.path
+    png.inc_dir = [os.path.join(prebuilt_dir, 'include').replace('\\', '/')]
+    png.found = True
+    jpeg = DEPS.add('JPEG', 'jpeg', ['SDL2_image-[2-9].*', 'jpeg-[6-9]*'], r'(lib){0,1}jpeg[-0-9]*\.dll$',
+                   find_header='jpeglib\.h', find_lib='(lib)?jpeg[-0-9]*\.lib')
+    jpeg.path = imageDep.path
+    jpeg.inc_dir = [os.path.join(prebuilt_dir, 'include').replace('\\', '/')]
+    jpeg.found = True
 
     dllPaths = {
         'png': imageDep.path,

--- a/buildconfig/download_win_prebuilt.py
+++ b/buildconfig/download_win_prebuilt.py
@@ -78,6 +78,10 @@ def get_urls(x86=True, x64=True, sdl2=True):
             'https://www.libsdl.org/projects/SDL_mixer/release/SDL2_mixer-devel-2.0.4-VC.zip',
             '9097148f4529cf19f805ccd007618dec280f0ecc',
             ],
+            [
+            'https://www.ijg.org/files/jpegsr9c.zip',
+            '9ca086c960ffc4bff821ba194c8d0a15f69eae09',
+            ],
             # [
             #  'https://www.libsdl.org/release/SDL2-2.0.9-win32-x86.zip',
             #  '04a48d0b429ac65f0d9b33bd1b75d77526c0cccf'
@@ -217,6 +221,24 @@ def place_downloaded_prebuilts(temp_dir, move_to_dir, x86=True, x64=True, sdl2=T
     for prebuilt_dir in prebuilt_dirs:
         path = os.path.join(move_to_dir, prebuilt_dir)
         print("copying into %s" % path)
+
+        # update jpeg
+        for file in ('jerror.h', 'jmorecfg.h', 'jpeglib.h'):
+            shutil.copyfile(
+                os.path.join(
+                    temp_dir,
+                    'jpegsr9c',
+                    'jpeg-9c',
+                    file
+                ),
+                os.path.join(
+                    move_to_dir,
+                    prebuilt_dir,
+                    'include',
+                    file
+                )
+            )
+
         copy(
             os.path.join(
                 temp_dir,

--- a/src_c/imageext.c
+++ b/src_c/imageext.c
@@ -28,8 +28,6 @@
  */
 #include "pygame.h"
 
-#if IS_SDLv1
-
 /* Keep a stray macro from conflicting with python.h */
 #if defined(HAVE_PROTOTYPES)
 #undef HAVE_PROTOTYPES
@@ -52,8 +50,6 @@
 
 #include <jerror.h>
 #include <jpeglib.h>
-
-#endif /* IS_SDLv1 */
 
 #include "pgcompat.h"
 
@@ -316,6 +312,8 @@ SavePNG(SDL_Surface *surface, const char *file)
 #else  /* IS_SDLv2 */
     Uint8 surf_alpha = 255;
     Uint32 surf_colorkey;
+    int has_colorkey = 0;
+    SDL_BlendMode surf_mode;
 #endif /* IS_SDLv2 */
 
     ss_rows = 0;
@@ -380,7 +378,17 @@ SavePNG(SDL_Surface *surface, const char *file)
         SDL_SetAlpha(surface, 0, 255);
     if (surf_flags & SDL_SRCCOLORKEY)
         SDL_SetColorKey(surface, 0, surface->format->colorkey);
-#endif /* IS_SDLv1 */
+#else /* IS_SDLv2 */
+    SDL_GetSurfaceAlphaMod(surface, &surf_alpha);
+    SDL_SetSurfaceAlphaMod(surface, 255);
+    SDL_GetSurfaceBlendMode(surface, &surf_mode);
+    SDL_SetSurfaceBlendMode(surface, SDL_BLENDMODE_NONE);
+
+    if (SDL_GetColorKey(surface, &surf_colorkey) == 0) {
+        has_colorkey = 1;
+        SDL_SetColorKey(surface, SDL_FALSE, surf_colorkey);
+    }
+#endif /* IS_SDLv2 */
 
     ss_rect.x = 0;
     ss_rect.y = 0;
@@ -400,10 +408,10 @@ SavePNG(SDL_Surface *surface, const char *file)
     if (surf_flags & SDL_SRCCOLORKEY)
         SDL_SetColorKey(surface, SDL_SRCCOLORKEY, surf_colorkey);
 #else  /* IS_SDLv2 */
-    if (SDL_GetColorKey(surface, &surf_colorkey) == 0)
-        SDL_SetColorKey(ss_surface, SDL_TRUE, surf_colorkey);
-    SDL_GetSurfaceAlphaMod(surface, &surf_alpha);
+    if (has_colorkey)
+        SDL_SetColorKey(surface, SDL_TRUE, surf_colorkey);
     SDL_SetSurfaceAlphaMod(surface, surf_alpha);
+    SDL_SetSurfaceBlendMode(surface, surf_mode);
 #endif /* IS_SDLv2 */
 
     for (i = 0; i < ss_h; i++) {
@@ -472,7 +480,7 @@ j_empty_output_buffer(j_compress_ptr cinfo)
     dest->pub.next_output_byte = dest->buffer;
     dest->pub.free_in_buffer = OUTPUT_BUF_SIZE;
 
-    return TRUE;
+    return 1;
 }
 
 static void
@@ -552,9 +560,9 @@ write_jpeg(const char *file_name, unsigned char **image_buffer,
      */
 
     jpeg_set_defaults(&cinfo);
-    jpeg_set_quality(&cinfo, quality, TRUE);
+    jpeg_set_quality(&cinfo, quality, 1);
 
-    jpeg_start_compress(&cinfo, TRUE);
+    jpeg_start_compress(&cinfo, 1);
 
     /* try and write many scanlines at once.  */
     while (cinfo.next_scanline < cinfo.image_height) {
@@ -804,7 +812,6 @@ image_save_ext(PyObject *self, PyObject *arg)
         const char *name = Bytes_AS_STRING(oencoded);
         Py_ssize_t namelen = Bytes_GET_SIZE(oencoded);
 
-#if IS_SDLv1
         if ((namelen >= 4) &&
             (((name[namelen - 1] == 'g' || name[namelen - 1] == 'G') &&
               (name[namelen - 2] == 'e' || name[namelen - 2] == 'E') &&
@@ -841,24 +848,6 @@ image_save_ext(PyObject *self, PyObject *arg)
             result = -2;
 #endif /* ~PNG_H */
         }
-#else /* IS_SDLv2 */
-        if ((namelen >= 4) &&
-            (((name[namelen - 1] == 'g' || name[namelen - 1] == 'G') &&
-              (name[namelen - 2] == 'e' || name[namelen - 2] == 'E') &&
-              (name[namelen - 3] == 'p' || name[namelen - 3] == 'P') &&
-              (name[namelen - 4] == 'j' || name[namelen - 4] == 'J')) ||
-             ((name[namelen - 1] == 'g' || name[namelen - 1] == 'G') &&
-              (name[namelen - 2] == 'p' || name[namelen - 2] == 'P') &&
-              (name[namelen - 3] == 'j' || name[namelen - 3] == 'J')))) {
-            result = IMG_SaveJPG(surf, name, JPEG_QUALITY);
-        }
-        else if ((namelen >= 3) &&
-                 ((name[namelen - 1] == 'g' || name[namelen - 1] == 'G') &&
-                  (name[namelen - 2] == 'n' || name[namelen - 2] == 'N') &&
-                  (name[namelen - 3] == 'p' || name[namelen - 3] == 'P'))) {
-            result = IMG_SavePNG(surf, name);
-        }
-#endif /* IS_SDLv2 */
     }
     else {
         result = -2;

